src/main/java/com/google/plantasy/utils/GameUtils.java

```diff
@@ -42,32 +43,31 @@ public static Entity createGameEntity(String gameName, DatastoreService datastor
     
     // intializing game property values
     Entity gameEntity = new Entity("Game");
+    Game game = new Game(gameEntity);
     ArrayList<String> userIds = new ArrayList<String>();
     String quizQuestion = "";
     long quiz_timestamp = 0;
- 
-    gameEntity.setProperty("gameName", gameName);
-    gameEntity.setProperty("userIds", userIds);
-    gameEntity.setProperty("quizQuestion", quizQuestion);
-    gameEntity.setProperty("quiz_timestamp", quiz_timestamp);
     Key gameKey = gameEntity.getKey();
- 
+
+    game.setGameName(gameName);
+    game.setUserIds(userIds);
+    game.setQuizQuestion(quizQuestion);
+    game.setQuizTimestamp(quiz_timestamp);
+
     datastore.put(gameEntity);
 
     // getting the game id for the game id property
-    Entity entity = null;
     try{
-      entity = datastore.get(gameKey);
-
-      String key = KeyFactory.keyToString(gameKey);
-      entity.setProperty("gameId", key);
-      datastore.put(entity);
+      gameEntity = datastore.get(gameKey);
     }catch(EntityNotFoundException e){
       log.severe("EntityNotFoundException; game entity not found when adding game id in create game");
       return null;
     }
+    String key = KeyFactory.keyToString(gameKey);
+    game.setGameId(key);
+    datastore.put(gameEntity);
  
-    return entity;
+    return gameEntity;
```

Comment:
considering we have a nice wrapper class around game entity now, we should maybe return the Game rather than the Entity. and if we really want to return an entity, we should be calling game.GetGameEntity() instead

Reply:
Okay, that's done too + some changed places where that is called.